### DISCUSSION TITLE
"ready" event for virtualizer ?

### DISCUSSION THREAD

**Discussion by gsdev01** [TanStack/virtual#703] [Q&A]
Hi, 
as I need dimensions of items used in virtualizer for a spatial navigation, is there event that fired when virtualizer is ready ?
Because right now, if I set the focus after I initialized the virtualizer, the virtualized elements are not ready yet and focus is lost because they are not in the DOM yet.

Thanks for any help

**Comment by piecyk** [comment]
Hmm there is not such event, on instance we have range object when it's not null then items will be rendered.

**Reply by gsdev01** [reply]
Thanks for the reply @piecyk ,

I just checked out, but when range object is not null when the container with the ref attached is empty... 
So I can't set my focus based on this..

**Reply by gsdev01** [reply]
any other idea @piecyk ?

**Reply by piecyk** [reply]
One quick hack is to push it to next tick with setTimeout, if you could create an example then will have a look.

**Reply by gsdev01** [reply]
setTimeout trick works like a charm, should mention this somewhere in doc..
